vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/util/FunctionHelper.java

```diff
@@ -1222,56 +1219,42 @@ public static void printUTF8String(UTF8StringPointable stringp) {
-        String fName;
-        try {
-            fName = getStringFromPointable(stringp, bbis, di);
-        } catch (SystemException e) {
-            throw new HyracksDataException(e);
-        }
+            DataInputStream di, ArrayBackedValueStorage abvs, IParser parser) throws IOException {
+        String fName = getStringFromPointable(stringp, bbis, di);
         readInDocFromString(fName, bbis, di, abvs, parser);
     }
 
     public static void readInDocFromString(String fName, ByteBufferInputStream bbis, DataInputStream di,
-            ArrayBackedValueStorage abvs, IParser parser)
-                    throws NumberFormatException, JsonParseException, IOException {
-        int bufferSize = Integer.parseInt(System.getProperty("vxquery.buffer_size", "-1"));
+            ArrayBackedValueStorage abvs, IParser parser) throws IOException {
         Reader input;
         if (!fName.contains("hdfs:/")) {
             File file = new File(fName);
-
             if (file.exists()) {
-                if (bufferSize > 0) {
-                    input = new BufferedReader(new InputStreamReader(new FileInputStream(file)), bufferSize);
-                } else {
-                    input = new InputStreamReader(new FileInputStream(file));
-                }
+                input = new InputStreamReader(new FileInputStream(file));
                 parser.parse(input, abvs);
+            } else {
+                throw new FileNotFoundException(file.getAbsolutePath());
             }
-        }
-        //else check in HDFS file system
-        else {
-            HDFSFunctions hdfs = new HDFSFunctions(null, null);
-            FileSystem fs = hdfs.getFileSystem();
-            if (fs != null) {
-                try {
+        } else {
+            // else check in HDFS file system
+            try {
+                HDFSFunctions hdfs = new HDFSFunctions(null, null);
+                FileSystem fs = hdfs.getFileSystem();
+                if (fs != null) {
                     String fHdfsName = fName.replaceAll("hdfs:/", "");
                     Path xmlDocument = new Path(fHdfsName);
                     if (fs.exists(xmlDocument)) {
                         InputStream in = fs.open(xmlDocument).getWrappedStream();
-                        if (bufferSize > 0) {
-                            input = new BufferedReader(new InputStreamReader(in), bufferSize);
-                        } else {
-                            input = new InputStreamReader(in);
-                        }
+                        input = new InputStreamReader(in);
                         parser.parse(input, abvs);
                         in.close();
+                    } else {
+                        throw new FileNotFoundException(xmlDocument.getName());
                     }
                     fs.close();
-                } catch (IOException e) {
-                    throw new HyracksDataException(e);
                 }
+            } catch (IOException e) {
```

Comment:
You added a try catch, which I think should have been just an else with a throws statement.